handlers/main.yml

```diff
@@ -2,6 +2,10 @@
 - name: run riak_disk_tune
   command: /bin/bash /usr/local/bin/riak_disk_tune.sh
 
+- name: restart riak and ensure enabled
+  service: name=riak enabled=yes state=restarted
+  when: riak_init_system == 'system'
```

Comment:
Relevant note is that tasks/main.yml flushes handlers, so variable scope is more predictable in principle in this case